picasso/src/main/java/com/squareup/picasso/OkHttp3Downloader.java

```diff
@@ -15,76 +15,25 @@
  */
 package com.squareup.picasso;
 
-import android.content.Context;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.support.annotation.VisibleForTesting;
-import java.io.File;
 import java.io.IOException;
 import okhttp3.Cache;
 import okhttp3.Call;
-import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.Response;
 
 /** A {@link Downloader} which uses OkHttp to download images. */
-public final class OkHttp3Downloader implements Downloader {
+final class OkHttp3Downloader implements Downloader {
   @VisibleForTesting final Call.Factory client;
-  private final Cache cache;
-  private boolean sharedClient = true;
+  private final @Nullable Cache cache;
+  private final boolean sharedClient;
 
-  /**
-   * Create new downloader that uses OkHttp. This will install an image cache into your application
-   * cache directory.
-   */
-  public OkHttp3Downloader(final Context context) {
-    this(Utils.createDefaultCacheDir(context));
-  }
-
-  /**
-   * Create new downloader that uses OkHttp. This will install an image cache into the specified
-   * directory.
-   *
-   * @param cacheDir The directory in which the cache should be stored
-   */
-  public OkHttp3Downloader(final File cacheDir) {
-    this(cacheDir, Utils.calculateDiskCacheSize(cacheDir));
-  }
-
-  /**
-   * Create new downloader that uses OkHttp. This will install an image cache into your application
-   * cache directory.
-   *
-   * @param maxSize The size limit for the cache.
-   */
-  public OkHttp3Downloader(final Context context, final long maxSize) {
-    this(Utils.createDefaultCacheDir(context), maxSize);
-  }
-
-  /**
-   * Create new downloader that uses OkHttp. This will install an image cache into the specified
-   * directory.
-   *
-   * @param cacheDir The directory in which the cache should be stored
-   * @param maxSize The size limit for the cache.
-   */
-  public OkHttp3Downloader(final File cacheDir, final long maxSize) {
-    this(new OkHttpClient.Builder().cache(new Cache(cacheDir, maxSize)).build());
-    sharedClient = false;
-  }
-
-  /**
-   * Create a new downloader that uses the specified OkHttp instance. A response cache will not be
-   * automatically configured.
-   */
-  public OkHttp3Downloader(OkHttpClient client) {
-    this.client = client;
-    this.cache = client.cache();
-  }
-
-  /** Create a new downloader that uses the specified {@link Call.Factory} instance. */
-  public OkHttp3Downloader(Call.Factory client) {
+  OkHttp3Downloader(Call.Factory client, @Nullable Cache cache, boolean sharedClient) {
```

Comment:
Why @Nullable vs Cache.NONE?

Reply:
there is no more cache abstraction. it's just an LruCache now.

Reply:
it's okhttp3.Cache.